python/ray/util/annotations.py

```diff
@@ -0,0 +1,76 @@
+def PublicAPI(*args, **kwargs):
+    """Annotation for documenting public APIs.
+
+    Public APIs are classes and methods exposed to end users of Ray. You
+    can expect these APIs to remain backwards compatible even across major Ray
+    releases.
+
+    Args:
+        stability: Either "stable" for stable features or "beta" for APIs that
+            are intended to be public but still in beta.
+
+    Examples:
+        >>> @PublicAPI
+        >>> def func(x):
+        >>>     return x
+
+        >>> @PublicAPI(stability="beta")
+        >>> def func(y):
+        >>>     return y
+    """
+    if len(args) == 1 and len(kwargs) == 0 and callable(args[0]):
+        return PublicAPI(stability="stable")(args[0])
+
+    if "stability" in kwargs:
+        stability = kwargs["stability"]
+        assert stability in ["stable", "beta"], stability
+    elif kwargs:
+        raise ValueError("Unknown kwargs: {}".format(kwargs.keys()))
+    else:
+        stability = "stable"
+
```

Comment:
add `functools.wrap` so signature/name go through? https://docs.python.org/3/library/functools.html#functools.wraps

Reply:
This is actually unnecessary since we pass-thru the original function.